ffwd-reporter/src/test/java/com/spotify/metrics/ffwd/FastForwardReporterTest.java

```diff
@@ -214,4 +217,33 @@ public void shouldFinalReportAfterStopWithFlush() throws Exception {
 
         assert(counterBeforeStop < counterAfterStop);
     }
+
+    @Test
+    public void testKeyValuePrefixAddedOnce() throws Exception {
+        ArgumentCaptor<Metric> argumentCaptor = ArgumentCaptor.forClass(Metric.class);
+
+        doNothing().when(fastForward).send(argumentCaptor.capture());
+
+        MetricId name = MetricId.build("thename");
+
+        registry.meter(name);
+
+        reporter.start();
+
+        executorService.tick(REPORTING_PERIOD + REPORTING_PERIOD / 3, TimeUnit.MILLISECONDS);
+        verify(fastForward, atLeastOnce()).send(any(Metric.class));
+
+        List<Metric> metrics = argumentCaptor.getAllValues();
+
+        List<String> keys = new ArrayList<>();
+
+        List<String> expectedKeys = Arrays.asList("test.hi", "test.thename", "test.thename",
+        "test.thename", "test.hi", "test.thename", "test.thename", "test.thename");
+
+        for (Metric metric : metrics) {
+            keys.add(metric.getKey());
+        }
+
+        assertEquals(keys, expectedKeys);
```

Comment:
it might be more future-proof to do a comparison here where the order of metrics reported to the `reporter` doesn't matter, as I would guess it is dependent on how `report()` iterates over the metrics in the registry, something like:

```java
Set<String> expectedKeys = Set.of(...);
Set<String> actualKeys = argumentCaptor.getAllValues().stream().map(Metric::getKey).collect(Collectors.toSet());
assertEquals(expectedKeys, actualKeys);
```